packages/server-wallet/src/onchain-service/transaction-submission.ts

```diff
@@ -0,0 +1,120 @@
+import {providers, Wallet} from 'ethers';
+import _ from 'lodash';
+import PriorityQueue from 'p-queue';
+
+import {
+  TransactionSubmissionServiceInterface,
+  MinimalTransaction,
+  TransactionSubmissionOptions,
+} from './types';
+
+// FIXME: replace with
+// import {WalletError as ChannelWalletError} from '@statechannels/server-wallet';
+import {WalletError as ChannelWalletError} from '..';
+
+type Values<E> = E[keyof E];
+
+export class TransactionSubmissionError extends ChannelWalletError {
+  readonly type = ChannelWalletError.errors.OnchainError;
```

Comment:
TransactionSubmissionError?

Reply:
Sure, was going to reuse the type for all onchain services. Since this extends the `WalletError`, which has a static list of types, and the wallet is *only* concerned with an onchain transaction service my thinking was both services should use the same error type

Reply:
I think we may want to distinguish them as this code will not be inside `server-wallet` soon

Reply:
I'm thinking we may not really even want to have these errors extend the `WalletError` in that case, since changing the type means I always have to change the base class.

Is there anywhere that the code *expects* the error to be a `WalletError`?